apps/cowswap-frontend/src/modules/orders/containers/FulfilledOrderNotification/index.tsx

```diff
@@ -1,22 +1,45 @@
-import { useMemo } from 'react'
+import { useCallback, useMemo } from 'react'
 
 import { SupportedChainId } from '@cowprotocol/cow-sdk'
 
+import { EnhancedTransactionLink } from 'legacy/components/EnhancedTransactionLink'
+import { HashType } from 'legacy/state/enhancedTransactions/reducer'
 import { useOrder } from 'legacy/state/orders/hooks'
 
 import { parseOrder } from 'utils/orderUtils/parseOrder'
 
+import { OrderLinkWrapper } from '../../pure/commonStyled'
 import { ExecutedSummary } from '../ExecutedSummary'
 
 interface FulfilledOrderNotificationProps {
   uid: string
   chainId: SupportedChainId
+  onToastMessage(message: string): void
 }
-export function FulfilledOrderNotification({ chainId, uid }: FulfilledOrderNotificationProps) {
+export function FulfilledOrderNotification({ chainId, uid, onToastMessage }: FulfilledOrderNotificationProps) {
+  const ref = useCallback(
```

Comment:
Is this a ref or a callback? how do you make sure this is only triggered once and when the moment is right? Feels a bit hacky

Reply:
https://legacy.reactjs.org/docs/refs-and-the-dom.html#callback-refs

>React will call the ref callback with the DOM element when the component mounts, and call it with null when it unmounts. Refs are guaranteed to be up-to-date before componentDidMount or componentDidUpdate fires.